packages/remix-adapter/src/plugin.ts

```diff
@@ -0,0 +1,79 @@
+import type { Plugin, ResolvedConfig } from 'vite'
+import { mkdir, writeFile } from 'node:fs/promises'
+import { join, relative } from 'node:path'
+import { version, name } from '../package.json'
+const SERVER_ID = 'virtual:netlify-server'
+const RESOLVED_SERVER_ID = `\0${SERVER_ID}`
+
+// The virtual module that is the compiled server entrypoint.
+const serverCode = /* js */ `
+import { createRequestHandler } from "@netlify/remix-adapter";
+import * as build from "virtual:remix/server-build";
+export default createRequestHandler({ build });
+`
+// This is written to the functions directory. It just re-exports
+// the compiled entrypoint, along with Netlify function config.
+function generateNetlifyFunction(server: string) {
+  return /* js */ `
+    export { default } from "${server}";
+    export const config = {
+      name: "Remix server handler",
+      generator: "${name}@${version}",
```

Comment:
Is this actually working? `@netlify/functions` types don't mention `name` or `generator` - https://github.com/netlify/functions/blob/cdf899c63519d623e858f4db9d04d55c99456f77/src/function/v2.ts#L7-L43

Reply:
In Next Runtime we do output .json file with metadata like that currently

Reply:
I guess same question applies to `excludedPath` in edge plugin

Reply:
I'm basing it on this: https://docs.netlify.com/edge-functions/create-integration/#generate-declarations

Reply:
...and https://docs.netlify.com/edge-functions/declarations/#declare-edge-functions-inline

Reply:
Didn't test edge functions - but lambda ones don't work without `.json` with `generator` and `name`

Reply:
as in - they work - just don't fields don't do anything - I needed `.json` file to have `displayName` + `generator` show up

Reply:
You can try `ntl build` locally and check `.netlify/functions/manifest.json`

For this:
```
export const config = {
  path: "/no-meta",
  name: "Test",
  generator: "custom",
};
```

I get in manifest
```
{
      "generator": "internalFunc",
}
```

and for .json meta I get
```
{
      "displayName": "Custom name",
      "generator": "@pieh/wat@0.0.0-alpha.0",
}